google/cloud/security/actions/action_config_validator.py

```diff
@@ -0,0 +1,202 @@
+
+        Returns:
+          dict: The loaded configuration dictionary.
+
+        Raises:
+          ConfigLoadError: If the config cannot be loaded.
+        """
+        with open(os.path.abspath(path), 'rb') as filep:
+            try:
+                config = yaml.safe_load(filep)
+            except ValueError as yaml_error:
+                raise ConfigLoadError(yaml_error)
+        return config
+
+    @staticmethod
+    def _load_actions(config):
+        """Loads the actions actions from the config.
+
+        Args:
+          config (dict): A dictionary configuration.
+
+        Returns:
+          Tuple: A tuple of the actions and errors.
+        """
+        action_ids = defaultdict(int)
+        actions = []
+        errors = []
+        for action in config.get('actions', []):
+            action_id = action.get('id')
+            if action_id is None:
+                errors.append(MissingRequiredActionField('id'))
+            action_ids[action_id] += 1
+            if action_ids[action_id] > 1:
+                errors.append(DuplicateActionIdError(action_id))
+            actions.append(action)
+        return actions, errors
+
+    @staticmethod
+    def _check_action_type(action):
+        """Validates the action type.
+
+        Args:
+          action (dict): An action dictionary.
+
+        Returns:
+          ActionTypeDoesntExist: If the action type doesn't exist.
+        """
+        action_id = action.get('id')
+        action_type = action.get('type')
+        if not action_type:
+            return EmptyActionType(action_id)
+        parts = action_type.split('.')
+        try:
+            module = importlib.import_module('.'.join(parts[:-1]))
+            _ = getattr(module, parts[-1])
+        except (AttributeError, ImportError):
+            return ActionTypeDoesntExist(action_type)
+        return None
+
+    @staticmethod
+    def _check_trigger(action):
+        """Validates the triggers.
+
+        Args:
+          action (dict): An action dictionary.
+
+        Returns:
+          Error: If the trigger type doesn't exist or the trigger is missing.
+        """
+        action_id = action.get('id')
+        action_triggers = action.get('triggers')
+        if not action_triggers:
+            return EmptyActionTrigger(action_id)
+        for trigger in action_triggers:
+            parts = trigger.split('.')
+            if not parts[0] == 'rules':
+                return TriggerDoesntExist(trigger)
+            # TODO: once the rules classes and the action classes are set up,
+            # this should be changed handle any valid type instead of straight
+            # replacement.
+            parts[0] = 'google.cloud.security.auditor.rules'
+            try:
+                module = importlib.import_module('.'.join(parts[:-1]))
+                _ = getattr(module, parts[-1])
+            except (AttributeError, ImportError):
+                return TriggerDoesntExist(trigger)
```

Comment:
If there are multiple triggers that don't exist in a single action, the caller will only get the first one. This should return a list of all missing triggers that extends errors in validate. Be sure to update test_check_trigger_errors to check this.